### PR TITLE
Merges changes from wpcom as of 26th of June, 2015.

### PR DESCRIPTION
**Pull request by zinigor**

### CODE REVIEW

File: modules/contact-form/grunion-contact-form.php [hunk 1]
```diff
@@ -1674,7 +1611,7 @@ function __construct( $attributes, $content = null, $form = null ) {
 			if ( empty( $id ) ) {
 				$id = sanitize_title_with_dashes( 'g' . $form_id . '-' . $unescaped_label );
 				$i = 0;
-				$max_tries = 24;
+				$max_tries = 99;
```
Comment (kraftbj) [inline modules/contact-form/grunion-contact-form.php]: Nothing that this is the only change in this file that should come downstream, IMO.

Comment (zinigor) [inline modules/contact-form/grunion-contact-form.php]: Yeah, this whole thing looks botched, tests are failing, will fix it. Thanks for taking a look!

**Comment by zinigor** [comment]
@jeherve thanks for noting the reCAPTCHA goof, I have removed that commit.

**Comment by dereksmart** [comment]
Could we please rebase this?  Most of it looks pretty good, but testing IRL is a pain, as it still throws 4.3 notices and I get some `headers already sent` errors when trying to post.  It's hard to tell if it's because it's so far behind current master, or there's actually something wrong with the code.

**Comment by zinigor** [comment]
@dereksmart I have rebased this PR a while ago, can you please take a look at it again?

**Comment by dereksmart** [comment]
This branch is too old.  Most of these changes are obsolete.  Going to re-open these dotcom changes in smaller PR's.

**Pull request closed without merging**